portal/static/js/psaTracker.js

```diff
@@ -156,13 +156,16 @@
             validateResult: function(val) {
                 var isValid = !(isNaN(val) || parseInt(val) < 0 || parseInt(val) > 9999);
                 if(!isValid) {
-                    this.addErrorMessage = this.i18next.t("Result must be a number.");
+                    this.addErrorMessage = this.i18next.t("Result must be a number and within valid range (less than 9999).");
                 } else {
                     this.addErrorMessage = "";
                 }
                 return isValid;
             },
             validateDate: function(date) {
+                if (!date || date.length <= 9) { //don't show validation error until enough characters are entered, just return false
```

Comment:
Do you still need? Doesn't the validation error only appears after the field loses focus?

Reply:
@ivan-c ivan yup, you are right, it does.  I am putting a check here to actually not show validation error until the user has entered sufficient characters.  i was seeing it even when i didn't enter any character.